PROTOSPEC.md

```diff
@@ -0,0 +1,62 @@
+# Specification outline
+
+An outline of the logic for this proposal, in the way that it would be specified:
+
+## Grammar
+
+```
+DecoratorIdentifier :: @ IdentifierName
+
+Decorator : DecoratorIdentifier ArgumentsList_opt
+
+DecoratorList : Decorator+
+```
+
+`DecoratorList`s can be used as in the current decorators proposal: decorating classes, methods, or field declarations.
+
+```
+DecoratorDeclaration :
+  decorator [no LineTerminator here] DecoratorIdentifier (FormalParameters[~Yield, ~Await])_opt { DecoratorList }
+```
+
+## Scopes
+
+Similar to `#names`, there is a parallel lexical scope for `@names`. This is important so that `with` statements do not provide a way to dynamically create decorators.
+
+Each construct which makes a new lexical scope for variables also names a lexical scope for decorators. The only type of declarations which decorators support are `const` declarations and `import { @name }` declarations.
```

Comment:
why only const? if they can be imported, they could be exported with `let` and rebound later, no?

Reply:
Oops, this needs to be updated to `decorator @name { }`. Anyway, all of these bindings are immutable to make them statically analyzable. They may be shadowed, and they may be renamed when importing.

Reply:
but i could `export let foo = @bar`, or would the only way to export a decorator be with syntax that would force it to be `const`?

Reply:
Nope, you couldn't do that. In the current draft, you can only define decorators as `decorator @foo { }`. In the previous draft, you could only use `const`, not `let`, and the name always has to begin with `@`, so `foo` would not be allowed.